### PR TITLE
[RUN-3079] record/verify resources.pak file sizes across macos architectures

### PR DESCRIPTION
**Pull request by toshok**
Add a step in `upload_build_artifacts.mjs` for all platforms to record/upload the size of the `resources.pak` file, as well as the list of all third_party deps (and their hashes).

Add a buildkite step to verify the third party deps and the sizes of the `resource.pak` files are the same between macos arm64/x86_64.

also track down and fix the remaining sources of differences between `resources.pak` on x86 vs arm (the `sync -D` and `//tools/license.py` commits).

### CODE REVIEW

File: components/resources/BUILD.gn [hunk 1]
```diff
@@ -107,6 +107,10 @@ action("about_credits") {
     "--target-os=$target_os",
     "--depfile",
     rebase_path(depfile, root_build_dir),
+    "--gn-out-dir",
+    ".",
+    "--gn-target",
+    "//chrome", # TODO(toshok) can't imagine we want this, but I'm not sure how else to do it?
```
Comment (toshok) [inline components/resources/BUILD.gn:113]: this is probably the sketchiest part of these changes.  Is there any way to figure out which toplevel target we're building?

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -141,6 +139,33 @@ steps:
       - "queue=runtime"
     artifact_paths:
       - "build_id/windows/x86_64/**"
+
+  - label: ":sleuth_or_spy: Verify cross-arch `.pak` sizes and dep hashes"
```
Comment (jazzdan) [inline .buildkite/pipeline.yml:143]: Excellent emoji choice

File: components/resources/BUILD.gn (hunk 1 above)
Comment (jazzdan) [reply components/resources/BUILD.gn:113]: I do not know, but `//chrome` seems like a safe bet for us?

**Review by jazzdan — APPROVED** [review]

File: components/resources/BUILD.gn (hunk 1 above)
Comment (toshok) [reply components/resources/BUILD.gn:113]: yeah, definitely my feeling

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -141,6 +139,33 @@ steps:
...
+    key: "verify-cross-arch-pak-sizes"
+    commands:
+      - "be_cmd download-macos-arm64-gclient_entries -- buildkite-agent artifact download build_id/macOS/arm64/entries ."
+      - "be_cmd download-macos-x86_64-gclient_entries -- buildkite-agent artifact download build_id/macOS/x86_64/entries ."
+      - "be_cmd macos-diff-gclient-entries -- diff -u build_id/macOS/arm64/entries build_id/macOS/x86_64/entries"
+      - "be_cmd download-macos-arm64-pak-sizes -- buildkite-agent artifact download build_id/macOS/arm64/pak-sizes ."
+      - "be_cmd download-macos-x86_64-pak-sizes -- buildkite-agent artifact download build_id/macOS/x86_64/pak-sizes ."
+      - "be_cmd macos-diff-pak-sizes -- diff -u build_id/macOS/arm64/pak-sizes build_id/macOS/x86_64/pak-sizes || true"
```
Comment (toshok) [inline .buildkite/pipeline.yml:151]: oh oops.  will remove this || true :)

**Pull request merged**